xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/WinVideoFilter.h

```diff
@@ -70,6 +71,16 @@ class CWinShader
   CD3DEffect   m_effect;
   CD3DTexture* m_target{ nullptr };
 
+protected:
+  static const float* getColorRange(bool isLimitedRange)
+  {
+    static const float fullRange[] = { 0.f, 1.f };
+    static const float limitedRange[] = { 16.f / 255.f, (235.f - 16.f) / 255.f };
+    static const float *ranges[] = { fullRange, limitedRange };
+
+    return ranges[isLimitedRange];
```

Comment:
casting a bool to int is unsafe, because a bool is anything not 0. How about `return isLimitedRange ? limitedRange : fullRange;`?